website/src/components/Submission/FileUpload/ColumnMapping.ts

```diff
@@ -0,0 +1,99 @@
+import { type ProcessedFile } from './fileProcessing';
+import type { InputField } from '../../../types/config';
+import stringSimilarity from '../../../utils/stringSimilarity';
+
+export class ColumnMapping {
+    private readonly map: ReadonlyMap<string, string | null>;
+
+    private constructor(map: ReadonlyMap<string, string | null>) {
+        this.map = map;
+    }
+
+    private static getBestMatchingTargetColumn(sourceColumn: string, inputFields: InputField[]): string | null {
+        const [bestMatch, score] = inputFields
+            .map((field): [string, number] => {
+                const score = Math.max(
+                    stringSimilarity(sourceColumn, field.name),
+                    stringSimilarity(sourceColumn, field.displayName ?? ''),
+                );
+                return [field.name, score];
+            })
+            .reduce((maxItem, currentItem) => (currentItem[1] > maxItem[1] ? currentItem : maxItem));
+        return score > 0.8 ? bestMatch : null;
+    }
+
+    /* Create a new mapping with the given columns, doing a best-effort to pre-match columns. */
+    public static fromColumns(sourceColumns: string[], inputFields: InputField[]) {
+        const mapping = new Map();
+        let availableFields = inputFields;
+        sourceColumns.forEach((sourceColumn) => {
+            const bestMatch = this.getBestMatchingTargetColumn(sourceColumn, availableFields);
+            mapping.set(sourceColumn, bestMatch);
+            availableFields = availableFields.filter((field) => field.name !== bestMatch);
```

Comment:
Hey! I found another issue here, so first you need to do an exact match and then only fuzzy matching, for example if I have my list of columns [geoLocAdmin3, geoLocAdmin1] then geoLocAdmin3 will get mapped to geoLocAdmin1 and geoLocAdmin1 will get mapped to geoLocAdmin2, see example below where I did this on the preview: 

![image](https://github.com/user-attachments/assets/07150d5b-754b-4ae9-8836-05a157c0657e)

It might also make sense to mark which fields have not been mapped exactly (e.g. score under 1) with a different color so that the user will quickly see that this mapping happened - otherwise they might miss sth being mapped incorrectly

Reply:
Ha! I had a murky feeling about the mapping, good think you spotted it. I've fixed it now. Also the non-exact matches are not in italic, does that work?

Reply:
I also added all the 'desired' flags!

Reply:
Thanks it looks great!